src/components/common/Button/index.tsx

```diff
@@ -0,0 +1,95 @@
+import styled, { ThemedStyledProps } from 'styled-components'
+import { ColourTheme, ThemeType } from 'styles/theme'
+
+const { MAIN, SECONDARY } = ThemeType
+
+export const ButtonBase = styled.button.attrs(({ $alt, $border, theme }: { // $alt button theme toggle
+  $alt?: boolean; $border?: boolean; theme: ThemedStyledProps<ColourTheme, ColourTheme> }) => ({
```

Comment:
That's a lot of types, make an interface out of them outside the component

Reply:
Will do